src/openeo_gfmap/manager/job_manager.py

```diff
@@ -83,16 +122,77 @@ def __init__(
 
         self._root_collection = self._normalize_stac()
 
+        # Add a property that calculates the number of maximum concurrent jobs
+        # dinamically depending on the time
+        self._dynamic_max_jobs = dynamic_max_jobs
+        self._max_jobs_worktime = max_jobs_worktime
+        self._max_jobs = max_jobs
+
+    def add_backend(
+        self,
+        name: str,
+        connection,
+        parallel_jobs: Optional[int] = 2,
+        dynamic_max_jobs: bool = False,
+        min_jobs: Optional[int] = None,
+        max_jobs: Optional[int] = None,
+    ):
+        if not dynamic_max_jobs:
+            if parallel_jobs is None:
+                raise ValueError(
+                    "When dynamic_max_jobs is set to False, parallel_jobs must be provided."
+                )
+            return super().add_backend(name, connection, parallel_jobs)
+
+        if min_jobs is None or max_jobs is None:
+            raise ValueError(
+                "When dynamic_max_jobs is set to True, min_jobs and max_jobs must be provided."
+            )
+
+        if isinstance(connection, Connection):
+            c = connection
+            connection = lambda: c  # noqa: E731
+        assert callable(connection)
+
+        # Create a new NamedTuple to store the dynamic backend properties
+        class _DynamicBackend(NamedTuple):
+            get_connection: Callable[[], Connection]
+
+            @property
+            def parallel_jobs(self) -> int:
+                current_time = datetime.now()
```

Comment:
with using datetime you will run into issues with different timezones. Ideally use:

from openeo.util rfc3339